convert.c

```diff
@@ -1840,6 +1840,9 @@ static int ident_filter_fn(struct stream_filter *filter,
 	struct ident_filter *ident = (struct ident_filter *)filter;
 	static const char head[] = "$Id";
 
+	if (gvfs_config_is_set(GVFS_BLOCK_FILTERS_AND_EOL_CONVERSIONS))
+		die("ident conversions not supported when running under GVFS");
```

Comment:
Why did this need to be added here when we are checking in `ident_to_git`?  There are a bunch of other `*_filter_fn` methods and why do we not need this code in them as well?

Reply:
The filter_fn does the conversion directly and doesn't call ident_to_git so it requires a separate test.  Good point on the other *_filter_fn functions though I only see one (lf_to_crlf_filter_fn) that actually does conversions that we need to block.  I'll submit a revised patch to fix that one as well.

Reply:
Can you help me understand what the other *_filter_fn methods are used for?
1. null_filter_fn
2. cascade_filter_fn
3. start_multi_file_filter_fn

Reply:
> null_filter_fn

This one simply copies the input to the output. No changes.

> cascade_filter_fn

This one takes two filters, feeds the input to the first, and the output of that one to the second filter. This is used when the `ident` filter is in effect *and* the worktree file should have CR/LF line endings.

> start_multi_file_filter_fn

Some conversions are really expensive, e.g. Git LFS' smudge. To help that, somebody who is now working at GitHub implemented the "multi_file_filter" where a smudge/clean filter can be declared as a "process" similar to our read object hook: it will be started, and then fed the files to convert one by one, rather than starting a separate process for each file individually. See edcc85814c87ebd7f3b1b7d3979fac3dfb84d308 for details.